scrapers/scraper.py

```diff
@@ -0,0 +1,268 @@
+
+        try:
+            oldFile = json.loads(open("../json/{}.json".format(language), "r").read())
+            for data in oldFile:
+                if data["name"] == pairName:
+                    history = data["history"]
+                    break
+                history = []
+        except FileNotFoundError:
+            history = []
+
+        for commit in commits:
+            if commit.startswith(' '):
+                continue
+            commitData = {}
+            lines = commit.split('\n')
+            for line in lines:
+                if line.startswith('commit'):
+                    commitData["sha"] = line[len("commit "):]
+                if line.startswith('Author'):
+                    commitData["author"] = line.split(": ")[1].split(" <")[0]
+                if line.startswith('Date'):
+                    commitData["date"] = line[len("date:"):].strip()
+
+            if any(commitData["sha"] == cm["sha"] for cm in history):
+                continue
+
+            dixFile = "https://raw.githubusercontent.com/apertium/apertium-{0}/{1}/apertium-{0}.{0}.dix".format(pairName, commitData["sha"])
+            stems = countDixStems(dixFile, True)
+            if stems == -1:
+                dixName = iso3to2[pairList[0]] + "-" + iso3to2[pairList[1]]
+                dixFile = "https://raw.githubusercontent.com/apertium/apertium-{0}/{2}/apertium-{1}.{1}.dix".format(pairName, dixName, commitData["sha"])
+                stems = countDixStems(dixFile, True)
+                if stems == -1:
+                    continue
+
+            commitData["stems"] = stems["stems"]
+            history.append(commitData)
+
+        langPackages.append({"name": pairName, "history": history})
+    return langPackages
+
+def monoData(packages, updatemailmap):
+    """Returns data for all monolingual dictionaries: state, stems, location and contributors"""
+    data = []
+    for package in packages:
+        if not(re.match(r"apertium-\w+$", package["name"]) and package["name"][len("apertium-"):] in languages):
+            continue
+
+        language = package["name"][len("apertium-"):]
+        stats = json.loads(requests.get("https://apertium.projectjj.com/stats-service/apertium-{}/Lexc".format(language)).text)["stats"]
+        for statistic in stats:
+            if statistic["stat_kind"] == "Stems":
+                stems = statistic["value"]
+                break
+        for topic in package["topics"]:
+            if topic[len("apertium-"):] in langLocations:
+                location = topic[len("apertium-"):]
+                break
+
+        subprocess.call('git clone https://github.com/apertium/apertium-{}'.format(language), stderr=subprocess.DEVNULL)
+        shutil.copyfile("./.mailmap", "./apertium-{}/.mailmap".format(language))
+        lines = subprocess.check_output('git log --format="%aN" --follow apertium-{0}.{0}.lexc'.format(language), cwd='./apertium-{}/'.format(language)).decode('utf-8')
```

Comment:
Are you running the script from the `scrapers` directory? Because this is the type of errors that happen when run from another directory.
Also, obviously this one fails if `git clone` fails because it has nowhere to run the command

Reply:
Ah no, this seems to be a Linux issue. But doing it with a list should work